mav_trajectory_generation/test/test_polynomial.cpp

```diff
@@ -0,0 +1,147 @@
+                        static_cast<double>(RAND_MAX)) +
+         min;
+}
+
+bool approxEqual(double x_1, double x_2) {
+  double dist = std::abs(x_1 - x_2);
+  return dist < kEqualityResolution;
+}
+
+TEST(MavTrajectoryGeneration, Convolution) {
+  Eigen::VectorXd coeffs_1(2), coeffs_2(2);
+  coeffs_1 << 1.0, 2.0;
+  coeffs_2 << -1.0, 3.0;
+  Polynomial p(coeffs_1), q(coeffs_2);
+  Polynomial convolution = p * q;
+  Eigen::VectorXd expected_convolution(3);
+  expected_convolution << coeffs_1(0) * coeffs_2(0),
+      coeffs_1(0) * coeffs_2(1) + coeffs_1(1) * coeffs_2(0),
+      coeffs_1(1) * coeffs_2(1);
+  CHECK_EIGEN_MATRIX_EQUAL(expected_convolution, convolution.getCoefficients());
+}
+
+TEST(PolynomialTest, FindMinMax) {
+  const double kTMin = -100.0;
+  const double kTMax = 100.0;
+  const double kCoeffMin = -100.0;
+  const double kCoeffMax = 100.0;
+
+  std::srand(1234567);
+  static int num_failures = 0;
+  const int kNumPolynomials = 1e2;
+  for (size_t i = 0; i < kNumPolynomials; i++) {
+    // Create random polynomial.
+    int num_coeffs = std::rand() % (Polynomial::kMaxN - 1) + 1;
+    Eigen::VectorXd coeffs(num_coeffs);
+    for (size_t i = 0; i < num_coeffs; i++) {
+      coeffs[i] = createRandomDouble(kCoeffMin, kCoeffMax);
+    }
+    Polynomial p(coeffs);
+
+    // Calculate minimum and maximum.
+    std::pair<double, double> min_sampling, max_sampling, min_computing,
+        max_computing;
+    const double t_start = createRandomDouble(kTMin, kTMax);
+    const double t_end = createRandomDouble(t_start, kTMax);
+    timing::Timer timer_sampling("find_min_max_sampling");
+    findMinMaxBySampling(p, t_start, t_end, &min_sampling, &max_sampling);
+    timer_sampling.Stop();
+    timing::Timer timer_analytic("find_min_max_analytic");
+    bool success = p.findMinMax(t_start, t_end, kDerivative, &min_computing,
+                                &max_computing);
+    timer_analytic.Stop();
+    if (!success) {
+      std::cout << "Failed to compute roots of derivative of polynomial: "
+                << coeffs.transpose() << std::endl;
+      num_failures++;
+      continue;
+    }
+    if (!approxEqual(max_sampling.first, max_computing.first) ||
+        !approxEqual(min_sampling.first, min_computing.first)) {
+      std::cout << "t_min_sampling: " << min_sampling.first << std::endl;
+      std::cout << "t_min_computing: " << min_computing.first << std::endl;
+      std::cout << "min_sampling: " << min_sampling.second << std::endl;
+      std::cout << "min_computing: " << min_computing.second << std::endl;
+
+      std::cout << "t_max_sampling: " << max_sampling.first << std::endl;
+      std::cout << "t_max_computing: " << max_computing.first << std::endl;
+      std::cout << "max_sampling: " << max_sampling.second << std::endl;
+      std::cout << "max_computing: " << max_computing.second << std::endl;
+    }
+    EXPECT_TRUE(approxEqual(max_sampling.first, max_computing.first));
```

Comment:
You can also just add the output above on EXPECT_TRUE() by just printing to expect true; then it outputs what you want when it fails.
```
EXPECT_TRUE(my_number == 3) << "I wish my number had been 3, but it was " << my_number << " instead."
```